File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TransitService.java
package com.skedgo.tripkit;

import com.google.gson.annotations.JsonAdapter;
import com.skedgo.tripkit.routing.RealTimeVehicle;
import com.skedgo.tripkit.routing.Shape;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@Gson.TypeAdapters
@JsonAdapter(GsonAdaptersTransitService.class)
@Value.Style(passAnnotations = JsonAdapter.class)
public interface TransitService {

    List<Shape> shapes();

    String realTimeStatus();

    RealTimeVehicle realtimeVehicle();

    List<RealTimeVehicle> realtimeAlternativeVehicle();

}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TransitService.kt
package com.skedgo.tripkit

import com.google.gson.annotations.JsonAdapter
import com.skedgo.tripkit.routing.RealTimeVehicle
import com.skedgo.tripkit.routing.Shape
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@JsonAdapter(
    GsonAdaptersTransitService::class
)
@Style(passAnnotations = [JsonAdapter::class])
interface TransitService {
    fun shapes(): List<Shape>

    fun realTimeStatus(): String

    fun realtimeVehicle(): RealTimeVehicle

    fun realtimeAlternativeVehicle(): List<RealTimeVehicle>
}